src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int DEFAULT_CAPACITY_GROWTH_MULTIPLIER = 2;
+
+    private int capacity = DEFAULT_CAPACITY;
+    private float loadFactor = DEFAULT_LOAD_FACTOR;
+    private int threshold;
+    private int capacityGrowthMultiplier = DEFAULT_CAPACITY_GROWTH_MULTIPLIER;
+    private int size;
+    private Node<K, V> [] table;
+
+    {
+        table = new Node[capacity];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkCapacity();
+        int index = computeIndex(key);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+        } else {
+            Node<K,V> node = table[index];
+            while (true) {
```

Comment:
`while (true)` constructs are a bit confusing when we read the code
let's put an exact condition here

(what do you think, would it be easier to add a new node to the beginning of linked list, and not to the end?)